docs/csharp/nullable-references.md

```diff
@@ -35,18 +37,57 @@ name!.Length;
 
-The nullable context is controlled using a new pragma:
+- `enable`: The nullable annotation context is **enabled**. The nullable warning context is **enabled**.
+  - Variables of a reference type, `string` for example, are non-nullable.  All nullability warnings are enabled.
+- `disable`: The nullable annotation context is **disabled**. The nullable warning context is **disabled**.
+  - Variables of a reference type are oblivious, just like earlier versions of C#. All nullability warnings are disabled.
+- `safeonly`: The nullable annotation context is **enabled**. The nullable warning context is **safeonly**.
+  - Variables of a reference type are nonnullable. All safety nullability warnings are enabled.
+- `warnings`: The nullable annotation context is **disabled**. The nullable warning context is **enabled**.
+  - Variables of a reference type are oblivious. All nullability warnings are enabled.
+- `safeonlywarnings`: The nullable annotation context is **disabled**. The nullable warning context is **safeonly**.
+  - Variables of a reference type are oblivious. All safety nullability warnings are enabled.
 
-- `#nullable enable` sets the nullable context to enabled.
-- `#nullable disable` sets the nullable context to disabled.
+You can also use directives to set these same contexts anywhere in your project:
+
+- `#nullable enable`: Sets the nullable annotation context and nullable warning context to **enabled**.
+- `#nullable disable`: Sets the nullable annotation context and nullable warning context to **disabled**.
+- `#nullable safeonly`: Set the nullable annotation context to **enabled**, and the warning context to **safeonly**.
+- `#nullable restore`: Restores the nullable annotation context and nullable warning context to the project settings.
+- `#pragma warning disable nullable`: Set the nullable warning context to **disabled**.
+- `#pragma warning enable nullable`: Set the nullable warning context to **enabled**.
+- `#pragma warning restore nullable`: Restores the nullable warning context to the project settings.
+- `#pragma warning safeonly nullable`: Sets the nullable warning context to **safeonly**.
+
+The default nullable annotation and warning contexts are `disabled`. That decision means that your existing code compiles without changes and without generating any new warnings.
+
+The differences between the `enabled` and `safeonly` nullable warning contexts are warnings for assigning a nullable reference to a non-nullable reference. The following assignment generates a warning in an `enabled` warning context, but not a `safeonly` warning context. However, the second line, where `s` is dereferenced, generates a warning in a `safeonly` context:
+
+```csharp
+string s = null; // warning when context is enabled.
```

Comment:
It might be worth explicitly saying it's the *warning* context here and in the next line, given that there are two different contexts that could be referred to.